SpringProject/src/it/java/com/gradproject2019/userConference/UserConferenceControllerIT.java

```diff
@@ -59,75 +61,120 @@ public void shouldReturn200AndSaveInterestInConference() {
+    }
 
-        ResponseEntity<List<ConferenceResponseDto>> response = getConferenceByUserId(uri);
+    @Test
+    public void shouldReturn409AndNotSaveWhenConferenceAlreadyFavourited() {
+        ResponseEntity<ErrorEntity> response = saveInterestExpectingError(userConferenceRequestDto);
+
+        Assert.assertEquals(409, response.getStatusCodeValue());
+        Assert.assertEquals("Conference already favourited.", response.getBody().getMessage());
+    }
+
+    @Test
+    public void shouldReturn200AndListOfConferencesUserIsInterestedIn() {
+        ResponseEntity<List<ConferenceResponseDto>> response = getConferencesByUserId();
 
         Assert.assertEquals(200, response.getStatusCodeValue());
         Assert.assertEquals(savedConference.getName(), response.getBody().get(0).getName());
         Assert.assertEquals(savedConference.getId(), response.getBody().get(0).getId());
-
     }
 
     @Test
-    public void shouldReturn200AndEmptyListWhenUserNotInterestedInConference() throws URISyntaxException {
-
-        URI uri = new URI(baseUri + "/" + 2L);
+    public void shouldReturn200AndEmptyListWhenUserNotInterestedInConference() {
+        userConferenceRepository.deleteAll();
 
-        ResponseEntity<List<ConferenceResponseDto>> response = getConferenceByUserId(uri);
+        ResponseEntity<List<ConferenceResponseDto>> response = getConferencesByUserId();
 
         Assert.assertEquals(200, response.getStatusCodeValue());
-        Assert.assertEquals(response.getBody().size(), 0);
+        Assert.assertEquals(true, response.getBody().isEmpty());
+    }
 
+    @Test
+    public void shouldReturn401WhenUserRequestingConferenceNotLoggedIn() {
+        ResponseEntity<ErrorEntity> response = getConferencesByUserIdExpectingAuthError();
+
+        Assert.assertEquals(401, response.getStatusCodeValue());
+        Assert.assertEquals("User unauthorized to perform action.", response.getBody().getMessage());
     }
 
     @Test
-    public void shouldReturn401WhenUserNotLoggedIn() throws URISyntaxException {
+    public void shouldReturn204AndDeleteUserConference() {
+        ResponseEntity response = deleteUserConferences();
 
-        URI uri = new URI(baseUri + "/" + savedUser.getId());
+        Assert.assertEquals(204, response.getStatusCodeValue());
+        Assert.assertTrue(userConferenceRepository.findAll().isEmpty());
+    }
 
-        ResponseEntity<ErrorEntity> response = getConferenceByUserIdExpectingAuthError(uri);
+    @Test
+    public void shouldReturn401WhenUserRequestingDeleteNotLoggedIn() {
+        ResponseEntity<ErrorEntity> response = deleteUserConferencesExpectingAuthError();
 
         Assert.assertEquals(401, response.getStatusCodeValue());
+        Assert.assertEquals("User unauthorized to perform action.", response.getBody().getMessage());
+    }
+
+    @Test
+    public void shouldReturn404WhenUserConferenceNotFound() throws URISyntaxException {
+        URI uri = new URI(baseUri.toString() + "/2");
```

Comment:
Is there a reason that a conference with an ID of 2 doesn't exist?

If you're looking for a number that clearly won't ever be used, maybe INTEGER.MAX_VALUE would be better here.

Reply:
ah yeah, my bad